clarity/src/vm/functions/options.rs

```diff
@@ -124,6 +125,17 @@ fn eval_with_new_binding(
     let memory_use = bind_value.get_memory_use();
     env.add_memory(memory_use)?;
 
+    if *env.contract_context.get_clarity_version() >= ClarityVersion::Clarity2 {
+        if let CallableContract(trait_data) = &bind_value {
+            inner_context.callable_contracts.insert(
+                bind_name.clone(),
+                CallableData {
+                    contract_identifier: trait_data.contract_identifier.clone(),
+                    trait_identifier: trait_data.trait_identifier.clone(),
+                },
+            );
+        }
+    }
```

Comment:
I think this mean that `traits` can be bound through `match` as well. Does the typechecker allow that?

Reply:
Nevermind, I see that it does! Cool